python.cmake

```diff
@@ -104,11 +104,23 @@ MACRO(FINDPYTHON)
 
   IF(NOT CMAKE_VERSION VERSION_LESS "3.12")
 
-    IF(DEFINED PYTHON_EXECUTABLE OR DEFINED Python_EXECUTABLE)
+    IF(DEFINED PYTHON_EXECUTABLE OR
+        DEFINED Python_EXECUTABLE OR
+        DEFINED Python2_EXECUTABLE OR
+        DEFINED Python3_EXECUTABLE)
+
+      IF(NOT DEFINED PYTHON_EXECUTABLE)
+        IF(DEFINED Python_EXECUTABLE)
+          set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
+        ELSEIF(DEFINED Python2_EXECUTABLE)
+          set(PYTHON_EXECUTABLE ${Python2_EXECUTABLE})
+        ELSEIF(DEFINED Python3_EXECUTABLE)
+          set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
+        ENDIF()
+      ENDIF()
+
       IF(NOT DEFINED Python_EXCUTABLE)
         SET(Python_EXCUTABLE ${PYTHON_EXECUTABLE})
```

Comment:
That's not related to the PR but why `EXCUTABLE` instead of `EXECUTABLE` here?

Reply:
This is a mistake ;). My bad.

Reply:
thanks, let's fix